### DISCUSSION TITLE
Kitty terminfo on Openwrt

### DISCUSSION THREAD

**Discussion by Strykar** [kovidgoyal/kitty#3755] [Q&A] [answered]
OpenWRT doesn't package `tic`, so I compiled it for my release but it fails with:
```
$ infocmp -a xterm-kitty | ssh apu tic -x -o \~/.terminfo /dev/stdin
Error relocating /usr/bin/tic: _nc_set_writedir: symbol not found
Error relocating /usr/bin/tic: _nc_tic_expand: symbol not found
Error relocating /usr/bin/tic: _nc_check_termtype2: symbol not found
Error relocating /usr/bin/tic: _nc_write_entry: symbol not found
Error relocating /usr/bin/tic: _nc_syntax: symbol not found
Error relocating /usr/bin/tic: _nc_read_entry_source: symbol not found
Error relocating /usr/bin/tic: _nc_resolve_uses2: symbol not found
Error relocating /usr/bin/tic: _nc_write_object: symbol not found
Error relocating /usr/bin/tic: _nc_strict_bsd: symbol not found
Error relocating /usr/bin/tic: _nc_infotocap: symbol not found
Error relocating /usr/bin/tic: _nc_disable_period: symbol not found
Error relocating /usr/bin/tic: _nc_capcmp: symbol not found
Error relocating /usr/bin/tic: _nc_tic_written: symbol not found
Failed to compile terminfo using tic
```
What's causing this and how do I fix it?

**Comment by kovidgoyal** [comment, marked as answer]
tic on that machine is broken, as noted in the kitty FAQ in such cases you can copy over the compiled files from a similar linux system instead.

**Reply by Strykar** [reply]
> tic on that machine is broken, as noted in the kitty FAQ in such cases you can copy over the compiled files from a similar linux system instead.

That is how I have had it working in the past. I guess that's the solution moving forward.

**Comment by Luflosi** [comment]
The best solution would be to create a kitty-terminfo package that simply contains the xterm-kitty file from the tarball. Unfortunately I don't know how to create OpenWrt packages.

**Reply by Strykar** [reply]
> The best solution would be to create a kitty-terminfo package that simply contains the xterm-kitty file from the tarball. Unfortunately I don't know how to create OpenWrt packages.

I wasn't aware that it was platform architecture independent, I think this or copying it over might be the easiest solution.

**Comment by kovidgoyal** [comment]
On Wed, Jun 23, 2021 at 07:37:28AM -0700, Avinash H. Duduskar wrote:
> > The best solution would be to create a kitty-terminfo package that simply contains the xterm-kitty file from the tarball. Unfortunately I don't know how to create OpenWrt packages.
> 
> I wasn't aware that it was platform architecture independent, I think this or copying it over might be the easiest solution.

It's not entirely. hence you can use ones only from similar systems, and
that is why the instructions ask you to run tic instead of just copy the
binaries over.

**Reply by Luflosi** [reply]
So far I always just copied the terminfo file to all my OpenWrt routers and didn't notice any problems.
Do you think creating a package with the file from the tarball is a good idea or not?

**Reply by kovidgoyal** [reply]
I dont know anything about openwrt but assuming its based on a relatively normal linux system, yes it should be fine.

**Reply by Strykar** [reply]
It uses musl over glibc.

**Comment by Strykar** [comment]
Fixed the tic being broken issue:
```
root@apu:~# ldd `which tic`
	/lib/ld-musl-x86_64.so.1 (0x7fd5a549d000)
	libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x7fd5a5443000)
	libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x7fd5a542f000)
	libc.so => /lib/ld-musl-x86_64.so.1 (0x7fd5a549d000)
```
`/dev/stdin` is not symlinked on OpenWRT
```
$ infocmp -a xterm-kitty | ssh apu tic -x -o \~/.terminfo /proc/self/fd/0 
"/proc/self/fd/0", line 2, col 22, terminal 'xterm-kitty': older tic versions may treat the description field as an alias
```
And now..
`root@apu:~# cp .terminfo/x/xterm-kitty /usr/share/terminfo/x/`
fixes it.

And it appears there is no difference from the xterm-kitty file on Arch x64 and the one created above on Musl OpenWRT:
```
# diff ./xterm-kitty /usr/share/terminfo/x/xterm-kitty 
#
```
So just copy the file over folks :laughing:
